core/src/action.rs

```diff
@@ -50,13 +50,17 @@ pub struct Action {
 }
 
 impl Action {
+    /// immutable constructor from &Signal
```

Comment:
What do you mean by `immutable constructor`?

Reply:
the action is immutable, i'll just say constructor